tests/upgrades/test_hostgroup.py

```diff
@@ -16,155 +16,166 @@
-
-        # remove unset values before comparison
-        unset_keys = set(hostgroup_cloned) - set(hostgroup_origin)
-        for key in unset_keys:
-            del hostgroup_cloned[key]
-
-        # remove unique values before comparison
-        uniqe_keys = ('updated_at', 'created_at', 'title', 'id', 'name')
-        for key in uniqe_keys:
-            del hostgroup_cloned[key]
-        self.assertDictContainsSubset(hostgroup_cloned, hostgroup_origin)
-
-        # Delete hostgroup
-        hostgrp[0].delete()
-        with self.assertRaises(HTTPError):
-            hostgrp[0].read()
+    proxy = entities.SmartProxy().search(
+        query={'search': f'url = https://{settings.server.hostname}:9090'}
+    )[0]
+
+    # org = entities.Organization().create()
+    org = entities.Organization(name=request.node.name + "_org").create()
+    loc = entities.Location(organization=[org], name=request.node.name + "_loc").create()
+    parent_hostgroup = entities.HostGroup(
+        location=[loc.id], organization=[org.id], name=request.node.name + '_parent_host_grp'
+    ).create()
+    lc_env = entities.LifecycleEnvironment(
+        name=request.node.name + "_lce", organization=org
+    ).create()
+
+    domain = entities.Domain(name=request.node.name + "_domain").create()
+    architecture = entities.Architecture().create()
+    ptable = entities.PartitionTable().create()
+    operatingsystem = entities.OperatingSystem(
+        architecture=[architecture], ptable=[ptable], name=request.node.name + "_os"
+    ).create()
+    medium = entities.Media(operatingsystem=[operatingsystem]).create()
+    subnet = entities.Subnet(
+        location=[loc], organization=[org], name=request.node.name + "_subnet"
+    ).create()
+
+    host_group = entities.HostGroup(
+        architecture=architecture,
+        domain=domain,
+        location=[loc.id],
+        medium=medium,
+        name=request.node.name + "_host_grp",
+        operatingsystem=operatingsystem,
+        organization=[org.id],
+        ptable=ptable,
+        puppet_proxy=proxy,
+        puppet_ca_proxy=proxy,
+        subnet=subnet,
+        parent=parent_hostgroup,
+        lifecycle_environment=lc_env,
+        content_source=proxy,
+        root_pass='rootpass',
+    ).create()
+    assert host_group.name == request.node.name + "_host_grp"
+
+
+@post_upgrade(depend_on=test_pre_create_hostgroup)
+def test_post_crud_hostgroup(request):
+    """Hostgroup is intact post upgrade and update/delete/clone hostgroup
+
+    :id: postupgrade-79958754-94b6-4bfe-af12-7d4031cd2dd2
+
+    :steps:
+
+        1. After upgrade, check hostgroup entities.
+        2. Update existing hostgroup with new entities
+        3. Clone hostgroup.
+        4. Delete hostgroup, parent hostgroup, cloned hostgroup, domain, subnet, os, loc, org.
+
+    :expectedresults: After upgrade
+        1- Hostgroup entities remain same.
+        2- Hostgroup entties update should work.
```

Comment:
```suggestion
        2- Hostgroup should be updated.
```

Reply:
Same for other expected results.

Reply:
There was one typo in the statement, here we are checking the host group entities(subnet, os, domain, etc) update, so I think a typo fix will be sufficient.